### PR TITLE
adds veilid SAO macropad

### PR DESCRIPTION
**Pull request by c0ldbru**
## Description

Adds veilid SAO macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rot13labs/veilid_sao/rules.mk
Comment (zvecr) [inline keyboards/rot13labs/veilid_sao/rules.mk:1]: This file should be removed.

File: keyboards/rot13labs/veilid_sao/info.json
Comment (zvecr) [inline keyboards/rot13labs/veilid_sao/info.json:1]: Please rename this file to `keyboard.json`

File: keyboards/rot13labs/veilid_sao/info.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        "default": {
+            "animation": "cycle_left_right"
+        },
+        "layout": [
+            {"flags": 4, "matrix": [0, 0], "x": 0,   "y": 0}
+        ],
+        "max_brightness": 100
+    },
+    "layouts": {
+        "LAYOUT_sao_1x1": {
```
Comment (zvecr) [inline keyboards/rot13labs/veilid_sao/info.json:43]:
```suggestion
        "LAYOUT_ortho_1x1": {
```

File: keyboards/rot13labs/veilid_sao/info.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        "driver": "ws2812",
+        "default": {
+            "animation": "cycle_left_right"
+        },
+        "layout": [
+            {"flags": 4, "matrix": [0, 0], "x": 0,   "y": 0}
+        ],
+        "max_brightness": 100
+    },
+    "layouts": {
```
Comment (zvecr) [inline keyboards/rot13labs/veilid_sao/info.json:42]:
```suggestion
    "community_layouts": ["ortho_1x1"],
    "layouts": {
```

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            _delay_ms(1000);
+            SEND_STRING("start https://veilid.com" SS_TAP(X_ENT));
+        }
+        break;
+    }
+    return true;
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_sao_1x1(QK_MACRO_0),
```
Comment (zvecr) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:41]:
```suggestion
  [0] = LAYOUT_ortho_1x1(QK_MACRO_0),
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/rot13labs/veilid_sao/rules.mk
Comment (c0ldbru) [reply keyboards/rot13labs/veilid_sao/rules.mk:1]: Sweet! That's gone!

File: keyboards/rot13labs/veilid_sao/info.json
Comment (c0ldbru) [reply keyboards/rot13labs/veilid_sao/info.json:1]: Updated!

**Comment by c0ldbru** [comment]
Just checking in to see if theres anything else needed for this?

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case QK_MACRO_0:
```
Comment (drashna) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:21]:
```suggestion
    case QK_USER_1:
```

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            _delay_ms(1000);
+            SEND_STRING("start https://veilid.com" SS_TAP(X_ENT));
+        }
+        break;
+    }
+    return true;
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_ortho_1x1(QK_MACRO_0),
```
Comment (drashna) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:41]:
```suggestion
  [0] = LAYOUT_ortho_1x1(QK_USER_1),
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            _delay_ms(1000);
+            SEND_STRING(SS_TAP(X_LGUI));
+            _delay_ms(1000);
+            SEND_STRING("terminal");
+            _delay_ms(1000);
+            SEND_STRING(SS_TAP(X_ENT));
+            _delay_ms(1000);
+            SEND_STRING("open https://veilid.com" SS_TAP(X_ENT));
+            _delay_ms(1000);
+            SEND_STRING("start https://veilid.com" SS_TAP(X_ENT));
```
Comment (fauxpark) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:33]:
```suggestion
            SEND_STRING("open https://veilid.com\n");
            _delay_ms(1000);
            SEND_STRING("start https://veilid.com\n");
```

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    switch (keycode) {
+    case QK_USER_1:
+        if (record->event.pressed) {            
+            SEND_STRING(SS_DOWN(X_LGUI) SS_TAP(X_SPACE) SS_UP(X_LGUI));
+            _delay_ms(1000);
+            SEND_STRING(SS_TAP(X_LGUI));
+            _delay_ms(1000);
+            SEND_STRING("terminal");
+            _delay_ms(1000);
+            SEND_STRING(SS_TAP(X_ENT));
```
Comment (fauxpark) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:29]:
```suggestion
            tap_code(KC_ENTER);
```

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+*/
+#include QMK_KEYBOARD_H
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case QK_USER_1:
+        if (record->event.pressed) {            
+            SEND_STRING(SS_DOWN(X_LGUI) SS_TAP(X_SPACE) SS_UP(X_LGUI));
+            _delay_ms(1000);
+            SEND_STRING(SS_TAP(X_LGUI));
```
Comment (fauxpark) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:25]:
```suggestion
            tap_code(KC_LEFT_GUI);
```

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case QK_USER_1:
+        if (record->event.pressed) {            
+            SEND_STRING(SS_DOWN(X_LGUI) SS_TAP(X_SPACE) SS_UP(X_LGUI));
```
Comment (fauxpark) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:23]:
```suggestion
            tap_code16(LGUI(KC_SPACE));
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c (hunk 6 above)
Comment (c0ldbru) [reply keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:33]: @fauxpark OOoooooo, thats awesome! I didn't realize that worked! Thank you!!

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+            tap_code16(LGUI(KC_SPACE));
+            _delay_ms(1000);
+            tap_code(KC_LEFT_GUI);
+            _delay_ms(1000);
+            SEND_STRING("terminal");
+            _delay_ms(1000);
+            tap_code(KC_ENTER);
+            _delay_ms(1000);
+            SEND_STRING("open https://veilid.com\n");
+            _delay_ms(1000);
```
Comment (drashna) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:32]:
```suggestion
            wait_ms(1000);
            tap_code(KC_LEFT_GUI);
            wait_ms(1000);
            SEND_STRING("terminal");
            wait_ms(1000);
            tap_code(KC_ENTER);
            wait_ms(1000);
            SEND_STRING("open https://veilid.com\n");
            wait_ms(1000);
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by c0ldbru** [comment]
@fauxpark @zvecr anything else needed for this PR?

**Comment by c0ldbru** [comment]
Just checking back in to see if anything else is needed on this?

**Review by fauxpark — APPROVED** [review]

**Comment by sycophantic** [comment]
What is needed to get this merged?

File: keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+            wait_ms(1000);
+            tap_code(KC_LEFT_GUI);
+            wait_ms(1000);
+            SEND_STRING("terminal");
+            wait_ms(1000);
+            tap_code(KC_ENTER);
+            wait_ms(1000);
+            SEND_STRING("open https://veilid.com\n");
+            wait_ms(1000);
+            SEND_STRING("start https://veilid.com\n");
```
Comment (drashna) [inline keyboards/rot13labs/veilid_sao/keymaps/default/keymap.c:33]: This entire block can be replaced with just one function. 

```suggestion
            SEND_STRING(SS_DOWN(X_LGUI) SS_TAP(X_SPACE) SS_UP(X_LGUI) SS_DELAY(1000) SS_TAP(X_LGUI) SS_DELAY(1000) "terminal" SS_DELAY(1000) SS_TAP(X_ENTER) SS_DELAY(1000) "open https://veilid.com\n" SS_DELAY(1000) "start https://veilid.com\n");
```

**Review by drashna — APPROVED** [review]

**Comment by c0ldbru** [comment]
@zvecr Any chance you can look this over so it can get merged?

**Pull request merged**